src/Microsoft.AspNetCore.Razor.Evolution/Resources.resx

```diff
@@ -120,6 +120,9 @@
   <data name="IRBuilder_PopInvalid" xml:space="preserve">
     <value>The '{0}' operation is not valid when the builder is empty.</value>
   </data>
+  <data name="MismatchedContentEncoding" xml:space="preserve">
+    <value>The provided encoding does not match the content's encoding.</value>
```

Comment:
@rynowak suggested avoiding explosion test cases. This one probably makes sense to test though since we're purposefully exploding in an unexpected scenario. Can also dd the names; that'd be useful 👍